index.js

```diff
@@ -13,20 +13,27 @@ function setMultipart (req, done) {
 }
 
 function attachToBody (options, req, reply, next) {
-  req[kMultipart] = true
+  if (req.raw[kMultipart] !== true) {
+    next()
+    return
+  }
 
   const consumerStream = options.onFile || defaultConsumer
   const body = { }
+  const fileCount = { }
   const mp = req.multipart((field, file, filename, encoding, mimetype) => {
-    body[field] = {
+    fileCount[field] = fileCount[field] || 0
+    body[field] = body[field] || []
+    body[field][fileCount[field]] = {
```

Comment:
I don’t understand this fileCount logic. I think only one of these it’s called at any given time, so we can just use .push.

Reply:
That just was first solution came in mind. I've replaced it with push, tests seems fine.